phoenix-core/src/main/java/org/apache/phoenix/util/IndexUtil.java

```diff
@@ -958,6 +958,13 @@ public static void setScanAttributesForIndexReadRepair(Scan scan, PTable table,
             // This index table must be being deleted. No need to set the scan attributes
             return;
         }
+        // MetaDataClient modifies the index table name for view indexes if the parent view of an index has a child
+        // view. This, we need to recreate a PTable object with the correct table name for the rest of this code to work
+        if (table.getName().getString().contains(QueryConstants.CHILD_VIEW_INDEX_NAME_SEPARATOR)) {
+            int lastIndexOf = table.getName().getString().lastIndexOf(QueryConstants.CHILD_VIEW_INDEX_NAME_SEPARATOR);
```

Comment:
Should this logic be encapsulated somewhere? Is this the only place where we need to get the index name from a view index?

Reply:
This is a workaround. I think we need to fix this issue in the Phoenix core later. I do not want to give the impression that it is a final fix.